### DISCUSSION TITLE
Color of button widget is not changed anymore

### DISCUSSION THREAD

**Discussion by jwill0182** [fyne-io/fyne#3627] [Q&A] [answered]
Hello,

I´ve previously made it to change the background color of a button `widget.NewButton`. It all worked out pretty well. Since yesterday , after executing `go get -u plus go mod tidy`, the background color of fyne buttons - also in new projects which I create  see example below - is not changed anymore.  

By running this code, I´m getting just white Buttons displayed. Hope that somebody can help me out. Thx. a lot! 

`package main

import (
	"image/color"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/canvas"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/layout"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Colored Button")
	//resize window
	w.Resize(fyne.NewSize(400, 400))
	// Let show out button on screen
	w.SetContent(
		container.NewVBox(
			red_button(),
			red_button(),
			red_button(),
			red_button(),
			// let show / display buttons
			blue_button(),
			green_button(),
		),
	)
	w.ShowAndRun() // show and run app
}

// first colored button
func red_button() *fyne.Container { // return type
	btn := widget.NewButton("Visit", nil) // button widget
	// button color
	btn_color := canvas.NewRectangle(
		color.NRGBA{R: 255, G: 0, B: 0, A: 255})
	// container for colored button
	container1 := container.New(
		// layout of container
		layout.NewMaxLayout(),
		// first use btn color
		btn_color,
		// 2nd btn widget
		btn,
	)
	// our button is ready
	return container1
}

// Blue button
// copy previous button code and change it
func blue_button() *fyne.Container { // return type
	btn := widget.NewButton("Visit", nil) // button widget
	// button color
	btn_color := canvas.NewRectangle(
		color.NRGBA{R: 05, G: 0, B: 255, A: 255})
	// container for colored button
	container1 := container.New(
		// layout of container
		layout.NewMaxLayout(),
		// first use btn color
		btn_color,
		// 2nd btn widget
		btn,
	)
	// our button is ready
	return container1
}

// Green button
// copy and paste previous button
func green_button() *fyne.Container { // return type
	btn := widget.NewButton("Visit", nil) // button widget
	// button color
	btn_color := canvas.NewRectangle(
		color.NRGBA{R: 0, G: 255, B: 0, A: 255})
	// container for colored button
	container1 := container.New(
		// layout of container
		layout.NewMaxLayout(),
		// first use btn color
		btn_color,
		// 2nd btn widget
		btn,
	)
	// our button is ready
	return container1
}`

**Comment by Jacalz** [comment, marked as answer]
This is to be expected as we changed the theme for buttons. We introduced more importance levels for the buttons to reduce the impact of this. See the upgrade documentation: https://developer.fyne.io/api/v2.3/upgrading.html#changes

**Comment by jwill0182** [comment]
After setting the importance of the button color is being displayed again. 

`install_bt.Importance = widget.HighImportance`

Apparently the used `file_dialog = dialog.NewFolderOpen()`  is also impacted. Any chance it relates to the upgrade as well?

**Reply by Jacalz** [reply]
> Apparently the used file_dialog = dialog.NewFolderOpen() is also impacted. Any chance it relates to the upgrade as well?

I'm not entirely sure what you mean. Would you mind clarifying? The `HighImportance` field existed before the v2.3.0 release.
